webapi/src/main/scala/org/knora/webapi/slice/resources/repo/service/ResourcesRepoLive.scala

```diff
@@ -63,19 +67,254 @@ object ResourcesRepoLive {
+      }.toList
+
+    private def buildDateValuePattern(v: DateValueInfo, valueIri: String): List[TriplePattern] =
+      List(
+        iri(valueIri)
+          .has(KB.valueHasStartJDN, literalOf(v.valueHasStartJDN))
+          .andHas(KB.valueHasEndJDN, literalOf(v.valueHasEndJDN))
+          .andHas(KB.valueHasStartPrecision, literalOf(v.valueHasStartPrecision.toString()))
+          .andHas(KB.valueHasEndPrecision, literalOf(v.valueHasEndPrecision.toString()))
+          .andHas(KB.valueHasCalendar, literalOf(v.valueHasCalendar.toString())),
+      )
+
+    private def buildStillImageFileValuePattern(v: StillImageFileValueInfo, valueIri: String): List[TriplePattern] =
+      List(
+        iri(valueIri)
+          .has(KB.internalFilename, literalOf(v.internalFilename))
+          .andHas(KB.internalMimeType, literalOf(v.internalMimeType))
+          .andHas(KB.dimX, literalOf(v.dimX))
+          .andHas(KB.dimY, literalOf(v.dimY))
+          .andHas(KB.originalFilename, v.originalFilename.map(literalOf))
+          .andHas(KB.originalMimeType, v.originalMimeType.map(literalOf)),
+      )
+
+    private def buildStillImageExternalFileValuePattern(
+      v: StillImageExternalFileValueInfo,
+      valueIri: String,
+    ): List[TriplePattern] =
+      List(
+        iri(valueIri)
+          .has(KB.internalFilename, literalOf(v.internalFilename))
+          .andHas(KB.internalMimeType, literalOf(v.internalMimeType))
+          .andHas(KB.externalUrl, literalOf(v.externalUrl))
+          .andHas(KB.originalFilename, v.originalFilename.map(literalOf))
+          .andHas(KB.originalMimeType, v.originalMimeType.map(literalOf)),
+      )
+
+    private def buildDocumentFileValuePattern(v: DocumentFileValueInfo, valueIri: String): List[TriplePattern] =
+      List(
+        iri(valueIri)
+          .has(KB.internalFilename, literalOf(v.internalFilename))
+          .andHas(KB.internalMimeType, literalOf(v.internalMimeType))
+          .andHas(KB.originalFilename, v.originalFilename.map(literalOf))
+          .andHas(KB.originalMimeType, v.originalMimeType.map(literalOf))
+          .andHas(KB.dimX, v.dimX.map(i => literalOf(i)))
+          .andHas(KB.dimY, v.dimY.map(i => literalOf(i)))
+          .andHas(KB.pageCount, v.pageCount.map(i => literalOf(i))),
+      )
+
+    private def buildOtherFileValuePattern(v: OtherFileValueInfo, valueIri: String): List[TriplePattern] =
+      List(
+        iri(valueIri)
+          .has(KB.internalFilename, literalOf(v.internalFilename))
+          .andHas(KB.internalMimeType, literalOf(v.internalMimeType))
+          .andHas(KB.originalFilename, v.originalFilename.map(literalOf))
+          .andHas(KB.originalMimeType, v.originalMimeType.map(literalOf)),
+      )
+
+  }
 }
+
+extension (tp: TriplePattern)
+  def andHas(p: RdfPredicate, o: Option[RdfObject]): TriplePattern =
+    o.fold(tp)(o => tp.andHas(p, o))
+
+extension (iri: Iri)
+  def has(p: RdfPredicate, o: Option[RdfObject]): Option[TriplePattern] =
```

Comment:
I hope I have improved it in https://github.com/dasch-swiss/dsp-api/pull/3276/commits/b91fc161a4a9a00080a659321a7266c130a68a83 though I'm not sure I can fully eliminate the confusion.

> Also I think this would be helpful in other places as well.
> Why don't we put it in a separate file TriplePatternOps and make it easier available for the rest of the code base?

I was considering it, but then I didn't want to widen the scope of this PR any further. (If I move it to `TriplePatternOps` then I should also use it where applicable, I feel.)  
If you don't mind, I'd rather do that in a follow-up PR.

Reply:
The scaladoc helps for sure. Thanks.

> (If I move it to TriplePatternOps then I should also use it where applicable, I feel.)

Actually no, you just make it available to use. Integration to existing code can then happen later.